### ISSUE TITLE
[INSTALLATION/BUG] STEP 10 curl -A give source code of domain curl -I 520

### ISSUE THREAD

**Issue by Eckybrazzz** [mitchellkrogza/nginx-ultimate-bad-bot-blocker#269]
When testing setup as described following step 10 on curl -A other server gives pure html source. 

curl -I gives 520 error

Tested it on Ubuntu and Centos servers, origin server is Centos. And yes, changed it to the real domain
```
# curl -A "Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.96 Mobile Safari/537.36 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)" https://owndomain.com
<!DOCTYPE html><html class="no-js" lang="nl" prefix="og: http://ogp.me/ns#"><head><meta charset="UTF-8"> <!--[if IE ]><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"> <![endif]--><link rel="profile" href="http://gmpg.org/xfn/11" /><link rel="icon" href="https://owndomain.com/wp-content/uploads/2019/06/favicon.png"
```

```
# curl -A "Mozilla/5.0 (compatible; bingbot/2.0; +http://www.bing.com/bingbot.htm)" https://owndomain.com
<!DOCTYPE html><html class="no-js" lang="nl" prefix="og: http://ogp.me/ns#"><head><meta charset="UTF-8"> <!--[if IE ]><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"> <![endif]-->
```

```
# curl -I https://nl.mydomain.com -e http://100dollars-seo.com
HTTP/1.1 520 Origin Error
Date: Sat, 22 Jun 2019 23:41:38 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Set-Cookie: __cfduid=d02cd6523db929a66dc1c23516a0bc6021561246898; expires=Sun, 21-Jun-20 23:41:38 GMT; path=/; domain=.mydomain.com; HttpOnly; Secure
Expect-CT: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
Set-Cookie: cf_use_ob=0; path=/; expires=Sat, 22-Jun-19 23:42:08 GMT
Expires: Thu, 01 Jan 1970 00:00:01 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Server: cloudflare
CF-RAY: 4eb212f8bd4fc771-AMS
```

```
# curl -I https://nl.mydomain.com -e http://telegraaf.nl
HTTP/1.1 200 OK
Date: Sat, 22 Jun 2019 23:44:07 GMT
Content-Type: text/html; charset=utf-8
Connection: keep-alive
Set-Cookie: __cfduid=dbac1796cf36a568567a0374a89524d7e1561247047; expires=Sun, 21-Jun-20 23:44:07 GMT; path=/; domain=.mydomain.com; HttpOnly; Secure
Last-Modified: Sat, 22 Jun 2019 22:31:38 GMT
Vary: Accept-Encoding
X-Powered-By: centminmod
X-Xss-Protection: 1; mode=block
X-Content-Type-Options: nosniff
Expect-CT: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
Server: cloudflare
CF-RAY: 4eb216a1c9ecc84f-AMS
```

**Comment by Eckybrazzz** [comment]
Even made a test on a server where I did not install the Ultmimate bad bot blocker. hence it is a shared hosting.

```
# curl -A "Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.96 Mobile Safari/537.36 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)" https://somedomain.com
<!DOCTYPE html><html class="no-js" lang="nl-be" prefix="og: http://ogp.me/ns#"><head><meta charset="UTF-8"> <!--[if IE ]><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"> <![endif]--><link rel="profile" href="https://gmpg.org/xfn/11" /><link rel="icon" href="https://xn--bel
```

**Comment by mitchellkrogza** [comment]
Hi @Eckybrazzz that's what curl does by design, it's just like any web browser. So if it's a browser that is not blocked with the blocker, your web server does the very thing it was intended to do ie. pass the source code so the person can view your page.

The only way you can prevent curl from revealing source code is to block `cur'` with the custom includes but anyone can just right click "view source" or simply type `view-source:domain.com` into their address bar and see the same thing.

Your 2 `curl -A` commands both passed a good referrer so the expected response is to pass back the source code.

Your first `curl -I` passed a bad bot hence `520 Origin Error` and your second `curl -I` passed something that was not blocked hence `200 OK`

**Comment by mitchellkrogza** [comment]
Here's your same curl command against my server.

```
curl -A "Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.96 Mobile Safari/537.36 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)" https://[redacted].com```

```<!DOCTYPE html>
<!--[if IE 9 ]> <html lang="en-US" prefix="og: http://ogp.me/ns#" class="ie9 loading-site no-js"> <![endif]-->
<!--[if IE 8 ]> <html lang="en-US" prefix="og: http://ogp.me/ns#" class="ie8 loading-site no-js"> <![endif]-->
...............
..................
...............
.............
```

**Comment by Eckybrazzz** [comment]
Well, I used links for browsing, did not know that curl did the same. Not too old to learn something new. 

So, some troubleshooting to do.

But the strange thing, it worked before, but I was to lazy/happy and instead of a return 444; I did return 301 $scheme://www.facebook.com/YourPage; the to my business Facebook page and saw a huge increase of visitors. But removed that rule to the default return 444; again.

Afterwards I gonna buy you a coffee, you will stay awake for a long time :) 👍
